packages/veraswap-app/src/atoms/atoms.ts

```diff
@@ -163,10 +163,19 @@ export const hyperlaneGasPaymentAtom = atomWithQuery((get) => {
 export const transactionModalOpenAtom = atom<boolean>(false);
 export const transactionStepsAtom = atom<TransactionStep[]>([]);
 export const currentTransactionStepIdAtom = atom<string | undefined>(undefined);
-export const transactionHashesAtom = atom<{ swap?: string; bridge?: string; transfer?: string }>({});
+export const transactionHashesAtom = atom<{
```

Comment:
Make the object type a standalone type, eg `TransactionSteps`
and then you can do something like `TransactionStepKeys = keyof  TransactionSteps`
More robust, and that way for example with updateTransactionStepAtom, you can set the type of `update` to:
`update: {id: TransactionStepKeys; status (...)}`

or even better, define the keys first, as `type TransactionStepIds = "swap" | "bridge" (...)`
then , you can have as a type `Record<TransactionStepIds, string | undefined>`

you can also reuse this in your declaration of `TransactionSteps`